lms/djangoapps/instructor_analytics/basic.py

```diff
@@ -296,6 +299,44 @@ def extract_coupon(coupon, features):
     return [extract_coupon(coupon, features) for coupon in coupons_list]
 
 
+def list_problem_responses(course_key, problem_url):
+    """
+    Return responses to a given problem as a dict.
+
+    list_problem_responses(course_key, problem_url)
+
+    would return [
+        {'username': u'user1', 'state': u'...'},
+        {'username': u'user2', 'state': u'...'},
+        {'username': u'user3', 'state': u'...'},
+    ]
+
+    where `state` represents a student's response to the problem
+    identified by `problem_url`.
+    """
+    try:
+        problem_key = UsageKey.from_string(problem_url)
+        # Are we dealing with an "old-style" problem URL?
+        run = getattr(problem_key, 'run')
+        if not run:
+            problem_key = course_key.make_usage_key_from_deprecated_string(problem_url)
+        if problem_key.course_key != course_key:
+            raise InvalidKeyError(type(problem_key), problem_key)
+
+        smdat = StudentModule.objects.filter(
+            course_id=course_key,
+            module_state_key=problem_key
+        )
+        smdat = smdat.order_by('student')
+    except Exception:  # pylint: disable=broad-except
```

Comment:
I don't like this pattern. You're catching everything and just moving on. I think it would be better to explicitly catch the `InvalidKeyError` and let anything else bubble up; if something eventually goes wrong, we'd like to be able to debug it.